core/src/main/java/org/mule/runtime/core/api/event/CoreEvent.java

```diff
@@ -39,6 +42,21 @@
 @NoImplement
 public interface CoreEvent extends Serializable, Event {
 
+  /**
+   * @return a dummy event with no context and no values.
+   */
+  static CoreEvent nullEvent() {
+    return getNullEvent();
+  }
+
+  /**
+   * @param muleContext context on which the event will be associated.
+   * @return a dummy event associated on the given {@code muleContext} and no values.
+   */
+  static CoreEvent nullEvent(MuleContext muleContext) {
```

Comment:
is this one relying on the muleContext really needed in the api?

Reply:
It's actually the use case we have in the `mule-framework`.